irohad/main/iroha_conf_loader.hpp

```diff
@@ -28,6 +28,24 @@ namespace config_members {
   const char *MstSupport = "mst_enable";
 }  // namespace config_members
 
+static constexpr size_t kBadJsonPrintLength = 15;
+static constexpr size_t kBadJsonPrintOffsset = 5;
+static_assert(kBadJsonPrintOffsset <= kBadJsonPrintLength,
+              "The place of error is out of the printed string boundaries!");
+
+std::string reportJsonParsingError(const rapidjson::Document &doc,
+                                   const std::string &conf_path,
+                                   std::istream &input) {
+  const size_t error_offset = doc.GetErrorOffset();
+  const size_t print_offset =
+      std::max(error_offset, kBadJsonPrintOffsset) - kBadJsonPrintOffsset;
+  input.seekg(print_offset);
+  char json_error_buf[kBadJsonPrintLength + 1]; // includes room for trailing zero
+  json_error_buf[input.readsome(json_error_buf, sizeof(json_error_buf))] = 0;
```

Comment:
just `char json_error_buf[...] = {0};` above is enough

Reply:
also I think it's better just use std::string & dynamically allocate based on error_offset

Reply:
AFAIK string is not guaranteed to store a contiguous buffer internally, and using something like `std::copy` to insert chars one by one seems even more weird than using a `char[]`. And the zero-init syntax seems more readable, thanks for that, although it will set the whole buffer to zeros, and not just the last byte.

Reply:
> The elements of a basic_string are stored contiguously

https://en.cppreference.com/w/cpp/string/basic_string

Reply:
You are right, but an interesting fact is that before c++11 it was not guaranteed to store the characters contiguously.